app/src/main/java/org/fossasia/openevent/app/utils/CurrencyUtils.java

```diff
@@ -57,33 +58,43 @@ private CurrencyUtils() {
             .subscribeOn(Schedulers.computation());
     }
 
-    @SuppressWarnings({ "PMD.AvoidCatchingGenericException", "PMD.EmptyCatchBlock" })
-    public static List<String> getCurrencyCodesList() {
-        List<String> currencyCodes = new ArrayList<>();
-        List<String> priorCurrencyCodes = new ArrayList<>();
-
-        priorCurrencyCodes.add(Currency.getInstance(Locale.US).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.UK).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.GERMANY).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.JAPAN).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.CANADA).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.CHINA).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.KOREA).toString());
-
+    @SuppressWarnings({ "PMD.UseConcurrentHashMap"})
+    public static Map<String, String> getCountryCurrencyMap() {
+        Map<String, String> countryCurrencyMap = new TreeMap<>();
         for (Locale locale : Locale.getAvailableLocales()) {
             try {
                 Currency currency = Currency.getInstance(locale);
-                if (currency != null && !currencyCodes.contains(currency.toString())) {
-                    currencyCodes.add(currency.toString());
+                String country = locale.getDisplayCountry();
+                if (currency != null && country != null && !country.equals("")) {
+                    countryCurrencyMap.put(country, currency.toString());
                 }
-            } catch (Exception e) {
-                // No action
+            } catch (IllegalArgumentException e) {
+                Timber.d(e);
             }
         }
+        return countryCurrencyMap;
+    }
+
+    public static List<String> getCurrencyCodesList() {
+        Map<String, String> countryCurrencyMap = getCountryCurrencyMap();
+        ArrayList<String> currencyCodes = new ArrayList<>();
+
+        for (Map.Entry<String, String> entry : countryCurrencyMap.entrySet()) {
+            currencyCodes.add(entry.getValue());
+        }
+
+        return currencyCodes;
+    }
+
+    public static List<String> getCountryList() {
+        Map<String, String> countryCurrencyMap = getCountryCurrencyMap();
+        ArrayList<String> countries = new ArrayList<>();
+
+        for (Map.Entry<String, String> entry : countryCurrencyMap.entrySet()) {
+            countries.add(entry.getKey());
+        }
 
-        Collections.sort(currencyCodes);
-        priorCurrencyCodes.addAll(currencyCodes);
-        return priorCurrencyCodes;
+        return countries;
```

Comment:
Why is there a need for 3 functions

Reply:
We need sorted `List<String>`s for both spinners. We can create them using `List<Value> list = new ArrayList<Value>(map.values());` or `map.entrySet()`

But should be noted that the order of values returned by those methods is not defined and for HashMap and similar un-sorted Map implementations it will be effectively random.
So every time we need `currencyCodesList` or `CountriesList`, we'll have to sort.

I do agree that the methods need modification to simply use `Collections.sort`. Please tell if there's any better solution.

Reply:
The thing is the calculation of these lists or maps is considered an expensive operation. This function was only called once previously. Now it's called many times and in form of three functions.

We can cache the result inside the utils class itself and return precomputed result instead. In order to do so, make the methods non static and store the precomputed result in private members. And add `@Inject` annotation on the constructor of the class after making it public

LinkedHashMap is sorted by the order of entry

Reply:
We iterate using the values from `#getAvailableLocales`, in which some countries don't follow alphabetical order. 
TreeMap, on the other hand, is sorted by keys.